### PR TITLE
parser, evaluator: support mysql builtin functions ln(), log(), log2(), log10()

### PR DESCRIPTION
**Pull request by hamo**
basic support for mysql builtin functions ln(), log(), log2() and log10()

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/2258)

### CODE REVIEW

File: evaluator/builtin_math.go [hunk 1]
```diff
@@ -66,6 +66,81 @@ func builtinCeil(args []types.Datum, ctx context.Context) (d types.Datum, err er
...
+
+	switch len(args) {
+	case 1:
+		x, err := args[0].ToFloat64(sc)
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+
+		if x <= 0 {
+			d.SetNull()
```
Comment (coocood) [inline evaluator/builtin_math.go]: `d` is null by default, we don't need to call `SetNull`, just return will do.

**Comment by coocood** [comment]
@hamo 
Please add type infer at `plan/typeinferer.go` and `plan/typeiferer_test.go`.

**Comment by tiancaiamao** [comment]
Good job!  Please also add  ln, log, log2, log10 to this table:
https://github.com/pingcap/tidb/blob/893a1573013bc3fc9acd68a3cdd54e452c41e690/parser/parser_test.go#L79

**Comment by hamo** [comment]
PTAL @tiancaiamao @coocood

**Comment by tiancaiamao** [comment]
LGTM

**Comment by shenli** [comment]
@hamo Thank your very much!

**Pull request merged**